lib/Optimizer/Transforms/StateDecomposer.cpp

```diff
@@ -0,0 +1,128 @@
+/****************************************************************-*- C++ -*-****
```

Comment:
```suggestion
/*******************************************************************************
```
nit: use the emacs tag in files ending in `.h`, not `.cpp`

Reply:
will update

Reply:
removed this file